File: xpilot/contrib/jxpmap/org/xpilot/jxpmap/MapFuel.java
package org.xpilot.jxpmap;

import java.io.PrintWriter;
import java.io.IOException;

public class MapFuel extends MapObject {


    public MapFuel () {
        this(0, 0);
    }

    
    public MapFuel (int x, int y) {
        super("fuel.gif", x, y, 35 * 64, 35 * 64);
    }


    public void printXml (PrintWriter out) throws IOException {
        out.print("<Fuel x=\"");
        out.print(getBounds().x + getBounds().width / 2);
        out.print("\" y=\"");
        out.print(getBounds().y + getBounds().height / 2);
        out.println("\"/>");
    }
}

File: xpilot/contrib/jxpmap/python/MapFuel.py
from MapObject import MapObject
class MapFuel(MapObject):
    def __init__(self, x = 0, y = 0, bmstore = None):
        MapObject.__init__(self, x, y, 35 * 64, 35 * 64, bmstore['fuel.gif'])

    def printXml(self, file):
        file.write('<Fuel x="%d" y="%d"/>\n' % (
                   self.getBounds().x + self.getBounds().width / 2,
                   self.getBounds().y + self.getBounds().height / 2))
